tests/integration/cmor/test_fix.py

```diff
@@ -0,0 +1,910 @@
+"""Integration tests for fixes."""
+
+import dask.array as da
+import numpy as np
+import pytest
+from cf_units import Unit
+from iris.aux_factory import HybridHeightFactory, HybridPressureFactory
+from iris.coords import AuxCoord, DimCoord
+from iris.cube import Cube, CubeList
+
+from esmvalcore.cmor.check import CheckLevels, CMORCheckError
+from esmvalcore.exceptions import ESMValCoreDeprecationWarning
+from esmvalcore.preprocessor import (
+    cmor_check_data,
+    cmor_check_metadata,
+    fix_data,
+    fix_metadata,
+)
+
+
+# TODO: remove in v2.12
+@pytest.fixture(autouse=True)
+def disable_fix_cmor_checker(mocker):
+    """Disable the CMOR checker in fixes (will be default in v2.12)."""
+    class MockChecker:
+
+        def __init__(self, cube):
+            self._cube = cube
+
+        def check_metadata(self):
+            return self._cube
+
+        def check_data(self):
+            return self._cube
+
+    mock = mocker.patch('esmvalcore.cmor.fix._get_cmor_checker')
+    mock.return_value = MockChecker
+
+
+class TestAutomaticFix:
```

Comment:
I know that classes can be used for grouping tests in a file, but since all tests in this file are part of this single class I don't really see the added benefit. What is the advantage of using a class here?

Reply:
No particular reason, I think it's just nicer to group all these tests in one class. In addition, having all fixtures available with `self` is also quite nice 👍

Reply:
I'm not sure if I agree, but it's up to you.